task4/src/main/scala/CsvTask.scala

```diff
@@ -0,0 +1,122 @@
+
+import java.io._
+import java.text.SimpleDateFormat
+
+import scala.language.reflectiveCalls
+import scala.collection.mutable.{ArrayBuffer, Map}
+
+import java.time.format.DateTimeFormatter
+
+
+
+object CsvTask extends  App {
+
+  using(io.Source.fromFile("201608-citibike-tripdata.csv")) { source =>
```

Comment:
Is that possible to split this 104-lines long procedure into several methods/functions?